ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala

```diff
@@ -22,27 +22,31 @@ package object v2 {
       events: List[AcsUpdateEvent]
   )
 
-  sealed trait AcsUpdateEvent extends Product with Serializable
+  sealed trait AcsUpdateEvent extends Product with Serializable {
+    def stakeholders: Set[Ref.Party]
+
+    def templateId: Ref.Identifier
+  }
 
   object AcsUpdateEvent {
 
     final case class Create(
         transactionId: TransactionId,
-        nodeId: AbsoluteNodeId,
+        eventId: EventId,
         contractId: Value.AbsoluteContractId,
         templateId: Ref.Identifier,
         argument: Value.VersionedValue[Value.AbsoluteContractId],
         // TODO(JM,SM): understand witnessing parties
-        stakeholders: List[Ref.Party],
+        stakeholders: Set[Ref.Party],
     ) extends AcsUpdateEvent
 
     final case class Archive(
         transactionId: TransactionId,
-        nodeId: AbsoluteNodeId,
```

Comment:
I guess we should remove `AbsoluteNodeId` completely?

Reply:
what do you mean? It's not used anymore in this file, is it?

Reply:
I meant to replace it with `EventId` and removing it completely from the codebase. I introduced it in the v2 index-api PR.

Reply:
it is replaced with `domain.EventId`